File: src/main/java/a/b/noride2/Noride2.java
package a.b.noride2;

import constant.Constant;
import a.b.noride2.enchantment.*;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

// The value here should match an entry in the META-INF/mods.toml file
@Mod(Constant.MODDatas.MOD_ID)
public class Noride2 {

    public Noride2() {
        // 注册附魔
        Wufaxiacheng.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        Qiangzhifeixing.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        Qiaochiqudong.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        Zidongyidong.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        Jiaohua.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        BuXiaCheng.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        ZhiNengXiaCheng.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        PiaoFu.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        QiangZhiPaTi.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        QiangZhiJiPao.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        QiangZhiYouYong.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        ChuangYi.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        ChuangHua.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
        ZiYouChuangXing.ENCHANTMENTS.register(Constant.FML.INSTANCE.getModEventBus());
    }
}

File: src/main/kotlin/a/b/noride2/noride2.kt
package a.b.noride2

import a.b.noride2.enchantment.Enchantments
import a.b.noride2.utils.infix.InfixFunction.addTo
import constant.Constant
import constant.Constant.FML.modEventBus
import net.minecraftforge.fml.common.Mod

// The value here should match an entry in the META-INF/mods.toml file
@Mod(Constant.MODDatas.MOD_ID)
class Noride2 {
    init {
        // 注册附魔
        Enchantments.Wufaxiacheng.ENCHANTMENTS addTo modEventBus
        Enchantments.Qiangzhifeixing.ENCHANTMENTS addTo modEventBus
        Enchantments.Qiaochiqudong.ENCHANTMENTS addTo modEventBus
        Enchantments.Zidongyidong.ENCHANTMENTS addTo modEventBus
        Enchantments.Jiaohua.ENCHANTMENTS addTo modEventBus
        Enchantments.BuXiaCheng.ENCHANTMENTS addTo modEventBus
        Enchantments.ZhiNengXiaCheng.ENCHANTMENTS addTo modEventBus
        Enchantments.PiaoFu.ENCHANTMENTS addTo modEventBus
        Enchantments.QiangZhiPaTi.ENCHANTMENTS addTo modEventBus
        Enchantments.QiangZhiJiPao.ENCHANTMENTS addTo modEventBus
        Enchantments.QiangZhiYouYong.ENCHANTMENTS addTo modEventBus
        Enchantments.ChuangYi.ENCHANTMENTS addTo modEventBus
        Enchantments.ChuangHua.ENCHANTMENTS addTo modEventBus
        Enchantments.ZiYouChuangXing.ENCHANTMENTS addTo modEventBus
        Enchantments.NotInWater.ENCHANTMENTS addTo modEventBus
        Enchantments.NotInLava.ENCHANTMENTS addTo modEventBus
        Enchantments.InWater.ENCHANTMENTS addTo modEventBus
        Enchantments.InLava.ENCHANTMENTS addTo modEventBus
        Enchantments.QiangZhiCFly.ENCHANTMENTS addTo modEventBus
    }
}